json_rpc/jsonmarshal.nim

```diff
@@ -110,6 +120,14 @@ proc fromJson[T](n: JsonNode, argName: string, result: var seq[T]) =
     fromJson(n[i], argName, result[i])
 
 proc fromJson[N, T](n: JsonNode, argName: string, result: var array[N, T]) =
+  when T is byte:
```

Comment:
same remark as above is valid here

Reply:
This one however is not as simple. Currently there are a lot of values (like hashes and addresses) which are passed around as json strings, but in nim they are essentially byte arrays. One way to mitigate it would be to change their nim representations to `stint.uint`s, but I find it suboptimal, because (a) semantically they are not numbers and (b) conversion from hex string to uints and back is a bit more involved than for byte arrays, and is not really needed.

I think the most proper way would be to use distinct types for that and to define `fromJaon` overloads for them, but it felt like too much of a change at this point.

Reply:
That is how it is done in nimbus rpc by the way: https://github.com/status-im/nimbus/blob/master/nimbus/rpc/hexstrings.nim

Reply:
It is possible here to trigger an assertion from byteutils.  More specifically this one: https://github.com/status-im/nim-byteutils/blob/master/byteutils.nim#L51
Just provide as value e.g. `0x0000000000`. Basically it should be checked for `len*2` instead of `len` I believe.

Next to that, other wrong values such as "0x00" will not return with a specific "hex string error" but also still do the array check. This gives a confusing error message.

There is also different behaviour between the array input and string input version, in that the array version excepts it when N is not reached. (could argue that is faulty behaviour for the array version)

Anyway, this can be fixed, I know. However, the point I'm trying to make is that these, at first sight small additions, can create complexities that are not needed for all the "users" of nim-json-rpc.